From 12bc1c06773dcd5a5d4f6eddc6df0a77241122d3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marc-Andr=C3=A9=20Lureau?= <marcandre.lureau@gmail.com>
Date: Thu, 15 May 2014 12:46:00 +0200
Subject: [PATCH 3/3] clipboard: check that clipboard request does not belong to remote

Check clipboard owner, to avoid cyclic dependency of clipboard requests.
---
 gtk/spice-gtk-session.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/gtk/spice-gtk-session.c b/gtk/spice-gtk-session.c
index eab7e2f..66035ad 100644
--- a/gtk/spice-gtk-session.c
+++ b/gtk/spice-gtk-session.c
@@ -813,6 +813,9 @@ static gboolean clipboard_request(SpiceMainChannel *main, guint selection,
     GtkClipboard* cb;
     int m;
 
+    g_return_val_if_fail(s->clipboard_by_guest[selection] == FALSE, FALSE);
+    g_return_val_if_fail(s->clip_grabbed[selection], FALSE);
+
     if (read_only(self))
         return FALSE;
 
-- 
1.7.1

